website/docs/guides/versioned-resources.markdown

```diff
@@ -0,0 +1,62 @@
+
+Ultimately, we plan to completely automate the generation of Terraform resources to cover the core Kubernetes API. Having a set of versioned schemas that more closely matches the Kubernetes API definition is going to make this easier to achieve and will enable us to add built-in support for new API versions much faster. 
+
+
+## What will happen to the resources without versions in the name?
+
+These resources will continue to be supported and maintained as is through to v3.0.0 of the provider, at which point they will be marked as deprecated and then subsequently removed in v4.0.0.
+
+
+## `v1` resources
+
+Resources suffixed with a major version number are considered to have stable APIs that will not change. These resources will be supported by the provider so long as the API version continues to be supported by the Kubernetes API, and likely for some time after it is deprecated and removed as there is often a long tail of migration as users of the provider continue to support legacy infrastructure. 
+
+While the API contract for these resources is assumed to be concrete, we will still accept changes to add additional attributes to these resources for configuring convenience features such as the `wait_for_rollout` attribute seen on resources such as `kubernetes_deployment`. Changes to these attributes should always be accompanied by deprecation warnings, state upgraders, and follow our typical [semantic versioning](https://www.terraform.io/docs/extend/best-practices/versioning.html#versioning-specification) scheme.
+
+
+## `beta` resources
+
+We will continue to bring support for API resources which reach `beta` however it is expected that the API contract for these resources can still change and so they should be used with some caution. When a `beta` API changes we will provide a state upgrader for the resource where possible. Refer to the Kubernetes API documentation on the use of [beta resources](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+
+
+## `alpha` resources
+
+We will continue our policy of not building support for `alpha` versioned resources into the provider. Please use the [kubernetes_manifest](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest) resource to manage those resources. 
+
+
+## How can I move a resource without a version to its versioned resource name?
+
+The simplest, non-destructive way to do this is to modify the name of the resource to include the version suffix. Then remove the old resource from state and import the resource under the versioned resource like so:
+
+```
+terraform state rm kubernetes_config_map.example
```

Comment:
Would `terraform state mv` not be an easier option in this case?

Reply:
I considered this but `terraform state mv` does not work because, even though the resources actually share the same schema, terraform considers them different types and so produces an error message if you try to use mv:

```
╷
│ Error: Invalid state move request
│
│ Cannot move kubernetes_config_map.example to kubernetes_config_map_v1.example: resource types don't match.
╵
```

This only works for the same resource type, e.g for changing the name or moving it into a module.